src/inet/TCPEndPoint.cpp

```diff
@@ -1197,12 +1197,43 @@ void TCPEndPoint::Free()
     if (err != CHIP_NO_ERROR)
         Abort();
 
-    // Release the Retain() that happened when the end point was allocated
-    // [on LwIP, the object may still be alive if DoClose() used the
-    // EndPointBasis::DeferredFree() method.]
+    // Release the Retain() that happened when the end point was allocated [on
+    // LwIP, the object may still be alive if DoClose() used the DeferredFree()
+    // method.]
     Release();
 }
 
+#if CHIP_SYSTEM_CONFIG_USE_LWIP
+void TCPEndPoint::DeferredFree(ReleaseDeferralErrorTactic aTactic)
```

Comment:
DeferredRelease need to be used for all Lwip implementations, we should move this back to SystemPool.

Reply:
Why ? We will have lots of potential usages which do not need `DeferredFree`, Only `TCPEndpoint` and `UDPEndpoint` need it.

Reply:
Because DeferredRelease is a variant of Release in case the owner want to keep the object after release until the end of current event loop. This is not TCPEndPoint, UDPEndpoint, EndPointBasis specific. Otherwise, each object need to use this feature will have to implement a copy of this function by itself in future.

Reply:
No, think it another way, it doesn't defer the release of the object, it is deferring the `Release` calls.

Reply:
What is the benefit to refactor this function out of SystemPool? What gains we get if each user of this feature need to implement a copy of this feature by itself.